src/components/kubectl/logs.ts

```diff
@@ -0,0 +1,164 @@
+import * as vscode from 'vscode';
+import { Kubectl } from '../../kubectl';
+import { selectContainerForPod, PodSummary, quickPickKindName } from '../../extension';
+import { isPod } from '../../kuberesources.objectmodel';
+import * as kuberesources from '../../kuberesources';
+import { ResourceNode } from '../../explorer';
+import * as yaml from 'js-yaml';
+import * as kubectlUtils from '../../kubectlUtils';
+import { host } from '../../host';
+
+export enum LogsDisplayMode {
+    Show,
+    Follow
+}
+
+/**
+ * Fetches logs for a Pod. Handles use cases for fetching pods
+ * from an open document, or from the current namespace.
+ */
+export async function logsKubernetes(
+    kubectl: Kubectl,
+    explorerNode: ResourceNode | undefined,
+    displayMode: LogsDisplayMode
+) {
+    if (explorerNode) {
+        return await getLogsForExplorerPod(kubectl, explorerNode, displayMode);
+    }
+
+    return logsForPod(kubectl, displayMode);
+}
+
+/**
+ * Fetch logs from a Pod, when selected from the Explorer.
+ */
+async function getLogsForExplorerPod(
+    kubectl: Kubectl,
+    explorerNode: ResourceNode,
+    displayMode: LogsDisplayMode
+) {
+    const namespace = explorerNode.namespace;
+    const podSummary = { name: explorerNode.id, namespace: namespace };
+    const container = await selectContainerForPod(podSummary);
```

Comment:
From here onwards this appears to be the same as `getLogsForPod` - could we just call that method?

Reply:
fixed